src/test/java/com/microsoft/sqlserver/jdbc/NTLMConnectionTest.java

```diff
@@ -0,0 +1,689 @@
+
+
+/**
+ * A class for testing basic NTLMv2 functionality.
+ */
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.xAzureSQLDW)
+@Tag(Constants.xAzureSQLDB)
+@Tag(Constants.xAzureSQLMI)
+public class NTLMConnectionTest extends AbstractTest {
+
+    // custom formatter that contains nothing but the log message only
+    private static class CustomFormatter extends Formatter {
+        @Override
+        public String format(LogRecord record) {
+            StringBuffer sb = new StringBuffer();
+            sb.append(record.getMessage());
+            return sb.toString();
+        }
+    }
+
+    private static SQLServerDataSource dsNTLMLocal = null;
+
+    // NTLM properties defined
+    private static boolean ntlmPropsDefined = false;
+
+    // updated connection strong with NTLM properties
+    private static String connectionStringNTLM = connectionString;
+
+    private static String serverFqdn;
+
+    private static java.util.logging.Logger ntlmLogger = java.util.logging.Logger
+            .getLogger("com.microsoft.sqlserver.jdbc.internals.NTLMAuthentication");
+
+    private static ByteArrayOutputStream loggerContent = new ByteArrayOutputStream();
+    private static StreamHandler handler = new StreamHandler(new PrintStream(loggerContent), new CustomFormatter());
+
+    @BeforeAll
+    public static void setUp() {
+        // remove all other logging and add our own handler
+        LogManager.getLogManager().reset();
+        ntlmLogger.setLevel(Level.FINER);
+        ntlmLogger.addHandler(handler);
+        handler.setLevel(Level.FINER);
+
+        // if these properties are defined then NTLM is desired, modify connection string accordingly
+        String domain = System.getProperty("domain");
+        String user = System.getProperty("userNTLM");
+        String password = System.getProperty("passwordNTLM");
+
+        if (null != domain) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "domain", domain);
+            ntlmPropsDefined = true;
+        }
+
+        if (null != user) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "user", user);
+            ntlmPropsDefined = true;
+        }
+
+        if (null != password) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "password", password);
+            ntlmPropsDefined = true;
+        }
+
+        if (ntlmPropsDefined) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "authenticationScheme", "NTLM");
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "integratedSecurity", "true");
+
+            dsNTLMLocal = new SQLServerDataSource();
+            updateDataSource(connectionStringNTLM, dsNTLMLocal);
+        }
+    }
+
+    /**
+     * Tests basic NTLM authentication.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testNTLMBasicConnection() throws SQLException {
+        if (ntlmPropsDefined) {

```

Comment:
If `ntlmPropsDefined` is not `true`, this test will be skipped, but in Maven results it will be marked as passed. It would be better if we find a way of reporting these tests as skipped (perhaps with a tag?) when NTLM is not supported.